### DISCUSSION TITLE
Strange behaviour magit and org(on-ipython)

### DISCUSSION THREAD

**Discussion by acastanedam** [magit/magit#5008] [Support] [answered]
Hi,

This is a very strange problem I have with org's configuration and magit,  it turns out if ob-ipython is removed from org configuration, magit commit does not work anymore:
```  1 git … commit --
hint: Waiting for your editor to close the file...
Waiting for Emacs...
*ERROR*: Symbol’s function definition is void: second
```
Also it is also to debug further as `toggle-debug-on-error` does not bring any information. Is there any suspicion what is causing the problem?

Thanks in advance for any help.

**Comment by tarsius** [comment, marked as answer]
You have a package installed which uses the `second` function. It shouldn't do that because that function is provided by the `cl` package, which was deprecated many years ago, in favor of `cl-lib`, which comes with `cl-second`. Additionally, that package fails to `(require 'cl)`.

To get going again add `(require 'cl)` to your init file, but then also determine the package that uses `second` and get it fixed. I would recommend using `cadr` instead of `[cl-]second`, for elisp streat cred.

**Comment by acastanedam** [comment]
Hi @tarsius, definitely it had to do with the old `cl` library, I was checking which packages were using it and solved it. Thanks for your detailed answer.
